src/hooks/ERC20UniswapPrice.ts

```diff
@@ -0,0 +1,155 @@
+import { Pool } from '@uniswap/v3-sdk'
+import { Token } from '@uniswap/sdk-core'
+import { useQuery } from 'react-query'
+import { BigNumber } from '@ethersproject/bignumber'
+import { Contract } from '@ethersproject/contracts'
+import { abi as IUniswapV3PoolABI } from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json'
+import { abi as IUniswapV3FactoryABI } from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Factory.sol/IUniswapV3Factory.json'
+
+import { readProvider } from 'constants/readProvider'
+import { readNetwork } from 'constants/networks'
+
+interface Immutables {
+  factory: string
+  token0: string
+  token1: string
+  fee: number
+  tickSpacing: number
+  maxLiquidityPerTick: BigNumber
+}
+
+interface State {
+  liquidity: BigNumber
+  sqrtPriceX96: BigNumber
+  tick: number
+  observationIndex: number
+  observationCardinality: number
+  observationCardinalityNext: number
+  feeProtocol: number
+  unlocked: boolean
+}
+
+type Props = {
+  tokenSymbol: string
+  tokenAddress: string
+}
+
+// https://docs.uniswap.org/protocol/reference/deployments
+const UNISWAP_V3_FACTORY_ADDRESS = '0x1F98431c8aD98523631AE4a59f267346ea31F984'
+const WETH_ADDRESS = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
+const UNISWAP_FEE = 10000
+const networkId = readNetwork.chainId
+const wadPrecision = 18
+
+export function useUniswapPriceQuery({ tokenSymbol, tokenAddress }: Props) {
```

Comment:
Most of this code comes from https://docs.uniswap.org/sdk/guides/fetching-prices

There's a few caching optimizations we can probably do if we're happy with this approach.

Reply:
want to include this link as a comment in this file?

Reply:
Totally, good call.

Reply:
https://github.com/jbx-protocol/juice-interface/pull/311/commits/32821cbcbc0f50a9eb0c8fe14740b46185b00df0